pkg/restapi/edv/operation/operations.go

```diff
@@ -235,6 +249,23 @@ func (vc *VaultCollection) createDocument(vaultID string, document StructuredDoc
 	return vault.Put(document.ID, documentJSON)
 }
 
+// This function can't tell if the value before being encoded was precisely 128 bits long.
+// This is because the byte58.decode function returns an array of bytes, not just a string of bits.
+// So the closest I can do is see if the decoded byte array is 16 bytes long,
+// however this means that if the original value was 121 bits to 127 bits long it'll still be accepted.
+func checkIfBase58Encoded128BitValue(id string) error {
+	decodedBytes := base58.Decode(id)
+	if len(decodedBytes) == 0 {
+		return errNotBase58Encoded
+	}
+
+	if len(decodedBytes) != 16 {
+		return errNot128BitValue
+	}
+
+	return nil
+}
+
```

Comment:
Should I worry about this? This entire check is just so that the server can let the client know when their provided value doesn't conform to the spec, but with the library I'm using there's an edge-case that I can't detect.